aspnetcore/signalr/get-started-signalr-core.md

```diff
@@ -19,65 +19,106 @@ This tutorial teaches the basics of building a real-time app using SignalR for A
-
 This tutorial demonstrates the following SignalR development tasks:
 
 > [!div class="checklist"]
-> * Create an ASP.NET Core web app.
+> * Create a SignalR on ASP.NET Core web app.
 > * Create a SignalR hub to push content to clients.
-> * Use the SignalR JavaScript library to send messages and display updates from the hub.
+> * Modify the `Startup` class and configure the app.
 
-## Prerequisites
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/signalr/get-started-signalr-core/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+# Prerequisites
 
 Install the following software:
 
+# [Visual Studio](#tab/visual-studio)
+
+* [.NET Core 2.1.0 Preview 1 SDK](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300-preview1) or later
+* [Visual Studio 2017](https://www.visualstudio.com/downloads/) version 15.6 or later with the **ASP.NET and web development** workload
+* [npm](https://www.npmjs.com/get-npm)
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
 * [.NET Core 2.1.0 Preview 1 SDK](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300-preview1) or later
-* [Visual Studio 2017](https://www.visualstudio.com/downloads/) version 15.6 or later with the ASP.NET and web development workload
+* [Visual Studio Code](https://code.visualstudio.com/download) 
+* [C# for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)
 * [npm](https://www.npmjs.com/get-npm)
 
+-----
+
 ## Create an ASP.NET Core project that hosts SignalR client and server
 
-1. Use the **File** > **New Project** menu option and choose **ASP.NET Core Web Application**. Name the project `SignalRChat`.
+# [Visual Studio](#tab/visual-studio)
+
+1. Use the **File** > **New Project** menu option and choose **ASP.NET Core Web Application**. Name the project *SignalRChat*.
 
   ![New Project dialog in Visual Studio](get-started-signalr-core/_static/signalr-new-project-dialog.png)
 
-2. Select **Web Application** to create a project using Razor Pages. Then select **Ok**. Be sure that **ASP.NET Core 2.1** is selected from the framework selector, though SignalR runs on older versions of .NET.
+2. Select **Web Application** to create a project using Razor Pages. Then select **OK**. Be sure that **ASP.NET Core 2.1** is selected from the framework selector, though SignalR runs on older versions of .NET.
 
   ![New Project dialog in Visual Studio](get-started-signalr-core/_static/signalr-new-project-choose-type.png)
 
-  The libraries that host SignalR server-side code are included in the project template. Install the client-side JavaScript separately with [npm](https://www.npmjs.com/).
+3. The libraries that host SignalR server-side code are included in the project template. Install the client-side JavaScript separately with [npm](https://www.npmjs.com/).
```

Comment:
There's a step needed before this, since the *package.json* file should exist before installing npm packages. I suggest something like the following text for that step:
> 1. Right-click the project in **Solution Explorer** > **Add** > **New Item** > **npm Configuration File**. Name the file *package.json*.